### DISCUSSION TITLE
Trivy marks all dependencies of subdirectories as 'dev dependency' for PNPM

### DISCUSSION THREAD

**Discussion by SemProvoost** [aquasecurity/trivy#7381] [Bugs]
### Description

Bug Summary: Trivy is incorrectly identifying subdirectory packages, which is listed as a dependency in the package.json of a subdirectory, as a dev-dependency.

## Steps to reproduce
(Adding steps to reproduce here to allow myself of using markdown)

### Project structure
/ (root directory)
├── package.json
├── pnpm-lock.yaml
├── pnpm-workspace.yaml
└── subdir/
    └── package.json

### Root package.json content:
```
{
    "dependencies": {
        "await-sleep": "^0.0.1"
    }
}
```

### Subdir package.json content:
```
{
    "dependencies": {
        "sleep-utils": "1.0.3"
    }
}
```

### pnpm-workspace.yaml content:
```
packages:
  - "subdir/**"
```

### pnpm-lock.yaml (as result of `pnpm i` on root)
```
lockfileVersion: '9.0'

settings:
  autoInstallPeers: true
  excludeLinksFromLockfile: false

importers:

  .:
    dependencies:
      await-sleep:
        specifier: ^0.0.1
        version: 0.0.1

  subdir:
    dependencies:
      sleep-utils:
        specifier: 1.0.3
        version: 1.0.3

packages:

  await-sleep@0.0.1:
    resolution: {integrity: sha512-H3X3eAxwGpeNIk/yvFOs8g7500Q1YvzrxjSC9TNgLGtjrMFxPwhDdcT34QNs2iGWpZ+5WKkMJdjDoYs+Sw+TaA==}

  sleep-utils@1.0.3:
    resolution: {integrity: sha512-uJW7WDHISE1zJIdvoIewcdmis3pBvJhM30rni2gH7fHhV1NkTWLKw3J6CPRFdg3h+rFChFHzAgbkCKUErd4s8Q==}

snapshots:

  await-sleep@0.0.1: {}

  sleep-utils@1.0.3: {}
```

## Running Trivy
Running Trivy on the root of the project (`trivy fs . -f json --include-dev-deps --list-all-pkgs`) results in 
```
{
  "SchemaVersion": 2,
  "CreatedAt": "2024-08-25T14:59:48.762656+02:00",
  "ArtifactName": ".",
  "ArtifactType": "filesystem",
  "Metadata": {
    "ImageConfig": {
      "architecture": "",
      "created": "0001-01-01T00:00:00Z",
      "os": "",
      "rootfs": {
        "type": "",
        "diff_ids": null
      },
      "config": {}
    }
  },
  "Results": [
    {
      "Target": "pnpm-lock.yaml",
      "Class": "lang-pkgs",
      "Type": "pnpm",
      "Packages": [
        {
          "ID": "await-sleep@0.0.1",
          "Name": "await-sleep",
          "Identifier": {
            "PURL": "pkg:npm/await-sleep@0.0.1",
            "UID": "63283cc86327e55d"
          },
          "Version": "0.0.1",
          "Licenses": [
            "ISC"
          ],
          "Relationship": "direct",
          "Layer": {}
        },
        {
          "ID": "sleep-utils@1.0.3",
          "Name": "sleep-utils",
          "Identifier": {
            "PURL": "pkg:npm/sleep-utils@1.0.3",
            "UID": "c649275e081911ba"
          },
          "Version": "1.0.3",
          "Dev": true,
          "Licenses": [
            "MIT"
          ],
          "Indirect": true,
          "Relationship": "indirect",
          "Layer": {}
        }
      ]
    }
  ]
}
```

As you can see, the `sleep-utils` package is marked as `"Dev": true,` while it is no dev-dependency.

### Desired Behavior

In my example, the `sleep-utils` package should not have been marked as dev-dependency

### Actual Behavior

In my example, the `sleep-utils` package has incorrectly been marked as dev-dependency

### Reproduction Steps

```bash
See 'Steps to reproduce' in the description as that field supports markdown. 😁
```

### Target

None

### Scanner

None

### Output Format

JSON

### Mode

None

### Debug Output

```bash
Not relevant here 😁
```

### Operating System

macOS Ventura 13.3.1

### Version

```bash
0.54.1
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
@DmitriyLewen Can you please take a look?

**Comment by DmitriyLewen** [comment]
Created #7386 for this bug
